src/Commands/Files/RemoveFile.cs

```diff
@@ -11,29 +11,24 @@ public class RemoveFile : PnPWebCmdlet
     {
         private const string ParameterSet_SERVER_Delete = "Delete by Server Relative";
         private const string ParameterSet_SITE_Delete = "Delete by Site Relative";
-        private const string ParameterSet_SERVER_Recycle = "Recycle by Server Relative";
-        private const string ParameterSet_SITE_Recycle = "Recycle by Site Relative";
 
         [Parameter(Mandatory = true, Position = 0, ValueFromPipeline = true, ParameterSetName = ParameterSet_SERVER_Delete)]
-        [Parameter(Mandatory = true, Position = 0, ValueFromPipeline = true, ParameterSetName = ParameterSet_SERVER_Recycle)]
         [ValidateNotNullOrEmpty]
         public string ServerRelativeUrl = string.Empty;
 
         [Parameter(Mandatory = true, Position = 0, ValueFromPipeline = true, ParameterSetName = ParameterSet_SITE_Delete)]
-        [Parameter(Mandatory = true, Position = 0, ValueFromPipeline = true, ParameterSetName = ParameterSet_SITE_Recycle)]
         [ValidateNotNullOrEmpty]
         public string SiteRelativeUrl = string.Empty;
 
-        [Parameter(Mandatory = false, ParameterSetName = ParameterSet_SERVER_Recycle)]
-        [Parameter(Mandatory = false, ParameterSetName = ParameterSet_SITE_Recycle)]
+        [Parameter(Mandatory = false)]
         public SwitchParameter Recycle;
 
         [Parameter(Mandatory = false)]
         public SwitchParameter Force;
 
         protected override void ExecuteCmdlet()
         {
-            if (!string.IsNullOrEmpty(ServerRelativeUrl))
+            if (ParameterSpecified(SiteRelativeUrl))
```

Comment:
Hi @milanholemans , it needs to be `ParameterSpecified(nameof(SiteRelativeUrl))`, otherwise it will throw error

Reply:
You are right, my mistake! 😊